webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -20,18 +20,47 @@ import { UtilsService } from '../utils.service';
 import { HttpService } from '../http/http.service';
 import { SnackBarService } from '../snack-bar.service';
 import { Observer } from 'rxjs/Observer';
-import { Http } from '@angular/http';
+import { Http, Headers } from '@angular/http';
 import * as _ from 'lodash';
 import { MatDialog } from '@angular/material';
 import { Angular2InjectionTokens } from 'pluginlib/inject-resources';
 import { MessageDuration } from "../message-duration";
+import { OverwriteDatasetComponent } from '../../shared/dialog/overwrite-dataset/overwrite-dataset.component';
 import * as monaco from 'monaco-editor'
 
 let stateCache = {};
 let lastFile;
+const WHITESPACE_256 = String.fromCharCode.apply(null, new Array(256).fill(0x20));
 //Unsupported DS types
 const unsupportedTypes: Array<string> = ['G', 'B', 'C', 'D', 'I', 'R'];
 
+function isContentValidForDataset(content: string[], datasetAttrs: DatasetAttributes): boolean | string {
+  //TODO: validation of record length that is aware of how DBCS will effect actual length
+  //FB must have exactly lrecl, VB must have no more than lrecl. content cant be undefined.
+  const MAX_CONTENT_LENGTH = 5242880;
+  if (!content) {
+    return 'No Content';
+  }
+  let isFixedRecordLength = datasetAttrs.recfm.recordLength === 'F';
```

Comment:
@DivergentEuropeans @bobadesakshi21 Just FYI: there is no need to pad the records with spaces when record format is F, because zss does padding automatically.

Reply:
Alright. I will remove the padding.